### PR TITLE
Feature/pull request

### PR DESCRIPTION
**Pull request by magwas**

### CODE REVIEW

File: rules.java [hunk 1]
```diff
@@ -19,13 +19,27 @@
 shippable:
 	mkdir -p shippable
 
+ifneq ($(IS_PULL_REQUEST),true)
```
Comment (szirbucz) [inline rules.java]: Avoid negative conditions

File: rules.java [hunk 2]
```diff
@@ -19,13 +19,27 @@
...
 	$(TOOLCHAINDIR)/tools/pullanalize
 
+createdocs: $(MODEL_BASENAME).compiled codedocs
+
+else #IS_PULL_REQUEST
+
+sonar:
+
+createdocs:
+	$(TOOLCHAINDIR)/tools/noPullRequest
```
Comment (szirbucz) [inline rules.java]: noPullRequest will run when $IS_PULL_REQUEST is true? Sounds weird.

File: tools/noPullRequest [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+normalized_branch=$(echo $HEAD_BRANCH|sed 'sA[/_]A.Ag')
+sonar_uri="https://sonarcloud.io/dashboard?id=$REPO_NAME&branch=$HEAD_BRANCH&resolved=false" 
+repo_uri="https://repo.kodekonveyor.com/$REPO_NAME/$(echo $PULL_REQUEST_REPO_FULL_NAME |sed 'sA/.*AA')/$normalized_branch"
+echo  "building pull request. Sonar and documentation creation are not executed."
```
Comment (szirbucz) [inline tools/noPullRequest]: If they are not executed, what is the meaning of the next 2 lines?

**Review by szirbucz — CHANGES_REQUESTED** [review]

File: tools/noPullRequest (hunk 3 above)
Comment (magwas) [reply tools/noPullRequest]: The URLs point to the artifacts created by the non-PR build of the same pull request.

File: rules.java (hunk 1 above)
Comment (magwas) [reply rules.java]: fixing it

File: rules.java (hunk 2 above)
Comment (magwas) [reply rules.java]: it was an ifNdef

Comment (szirbucz) [reply rules.java]: Add more meaningful name

**Review by szirbucz — CHANGES_REQUESTED** [review]

**Pull request closed without merging**